DOCSP-41243 Add docs release notes for v20240710 (#807)
mongodb/docs-app-services: source/release-notes/backend.txt

2024 Releases
-------------

.. _backend_20240710:

10 July 2024 Release
~~~~~~~~~~~~~~~~~~~~

- Edge Server

>   - The bootstrap progress percentage is now shown in the Edge Server status UI.

Author: Clarified with engineer, it's the local UI! Added link to docs, status and the local UI are all documented

- :ref:`Enabling automatic deployment <deploy-github>` allows App Services to automatically push a copy of your App to your designated repository, 
  making the current :guilabel:`Export your application and upload to GitHub` section in the Deployment UI unnecessary. Updated the description 

>   under the :guilabel:`Enable Automatic Deployment` section to make this behavior clear. 

Author: Ticket to update screenshot: https://jira.mongodb.org/browse/DOCSP-41587

- Deployment logs for your application now record the ``user_id`` for Admin API and CLI requests.

- Fixed bug where :ref:`match expressions <database-triggers-match-expression>` for Triggers were converted to incorrect formats if they contained a regular expression (``$regex``).

- App Services Function names can now include hyphens (-).

- When a free or shared Atlas cluster exceeds 500 collections, the max number of collections for these tiers, you will get an error message.

>   For more information on the operational limits of free or shared clusters, see the :atlas:`Free and Shared Cluster Operational Limits <reference/free-shared-limitations/#operational-limits>`.

Reviewer: this link 404s for me. needs a slash before `<reference/...` maybe?

Author: Fixed!

> - :ref:`Version metadata <deployment-history>` for :ref:`Partion Based Sync Apps <partition-based-sync>` will now be pruned for versions older than 60 days.

Reviewer: ```suggestion
- :ref:`Version metadata <deployment-history>` for :ref:`Partition-Based Sync Apps <partition-based-sync>` will now be pruned for deployment versions older than 60 days.
```
Not sure how helpful the version metadata link is; it was not immediately clear why I landed in a Deployment History section. Could possibly revise the link text to be more clear
(and do we really need to link to the PBS page?)

Author: Edited, let me know what you think! Think the PBS link is still necessary since the mode is still in use. Wish I could add a '(deprecated)` note, but it's not officially deprecated right?

> - Removed option to switch between :guilabel:`Sync Types` in the UI of the cluster configuration page, unless the cluster is associated with a Partion Based Sync app. 

Reviewer: ```suggestion
- Removed option to switch between :guilabel:`Sync Types` in the :guilabel:`Device Sync` configuration screen, unless the cluster is associated with a Partition-Based Sync app. 
```
this is in Sync config settings, not the cluster config

I also assume we need to update the Sync Settings (and possibly other pages) to reflect this UI update?

Author: Ticket to update sync settings details: https://jira.mongodb.org/browse/DOCSP-41588

> - Removed option to switch between :guilabel:`Sync Types` in the UI of the cluster configuration page, unless the cluster is associated with a Partion Based Sync app. 
>   Then, the option to switch between sync modes will be available, as we recommend switching to Flexible Sync. For more information on migrating to Flexible Sync,
>   see :ref:`<realm-sync-migrate-modes>`.

Reviewer: another suggested version 
```suggestion
- Removed the option to change from Flexible Sync to Partition-Based Sync mode. The :guilabel:`Sync Types` field in the 
:guilabel:`Device Sync` configuration settings now only displays if the cluster is associated with a Partition-Based Sync App. 
  Then, the option to switch sync modes will be left available, as we recommend :ref:`migrating to Flexible Sync <realm-sync-migrate-modes>`.
```

> - :ref:`Destructive or breaking changes <breaking_changes>` require restarting Sync. When making a destructive change 

Author: Good catch, thank you! Filed this ticket to add this error to the docs: https://jira.mongodb.org/browse/DOCSP-41579

  in the Device Sync configuration UI, a modal now pops up to allow you to terminate and restart Sync when confirming the change.

> - Fixed bug where :ref:`revoking all user's sessions <revoke-user-sessions>` did not immediately disconnect users.

Author: updated, as it's supposed to refer to a singular user

2024 Releases
-------------

.. _backend_20240710:

10 July 2024 Release
~~~~~~~~~~~~~~~~~~~~

- Edge Server

>   - You can now track the bootstrap progress percentage in the status section of the Edge Server Local UI. 
>     For more information on tracking the status of you Edge Server instance, see :ref:`<edge-server-status>`. 

Reviewer: ```suggestion
  - You can now track the bootstrap progress percentage in the status section of the Edge Server local UI. 
    For more information on tracking the status of your Edge Server instance, see :ref:`<edge-server-status>`. 
```
looks like it's lowercase "local" in the linked docs

- :ref:`Enabling automatic deployment <deploy-github>` allows App Services to automatically push a copy of your App to your designated repository, 

>   making the :guilabel:`Export your application and upload to GitHub` section in the :guilabel:`Deployment` screen unnecessary. Removed the section and 
>   updated the description under the :guilabel:`Enable Automatic Deployment` section to make this behavior clear. 

Reviewer: ```suggestion
  making the :guilabel:`Export your application and upload to GitHub` section in the :guilabel:`Deployment` screen unnecessary. The section is removed and 
the description under the :guilabel:`Enable Automatic Deployment` section is updated to make this behavior clear. 
```
reads a little weird
